File: software/examples/perl/example_threshold.pl
#!/usr/bin/perl

use Tinkerforge::IPConnection;
use Tinkerforge::BrickletLine;

use constant HOST => 'localhost';
use constant PORT => 4223;
use constant UID => 'XYZ'; # Change to your UID

# Callback subroutine for reflectivity reached callback
sub cb_reflectivity_reached
{
    my ($reflectivity) = @_;

    print "Reflectivity: $reflectivity\n";
}

my $ipcon = Tinkerforge::IPConnection->new(); # Create IP connection
my $l = Tinkerforge::BrickletLine->new(&UID, $ipcon); # Create device object

$ipcon->connect(&HOST, &PORT); # Connect to brickd
# Don't use device before ipcon is connected

# Get threshold callbacks with a debounce time of 1 second (1000ms)
$l->set_debounce_period(1000);

# Register reflectivity reached callback to subroutine cb_reflectivity_reached
$l->register_callback($l->CALLBACK_REFLECTIVITY_REACHED, 'cb_reflectivity_reached');

# Configure threshold for reflectivity "greater than 2000"
$l->set_reflectivity_callback_threshold('>', 2000, 0);

print "Press key to exit\n";
<STDIN>;
$ipcon->disconnect();

File: software/examples/ruby/example_threshold.rb
#!/usr/bin/env ruby
# -*- ruby encoding: utf-8 -*-

require 'tinkerforge/ip_connection'
require 'tinkerforge/bricklet_line'

include Tinkerforge

HOST = 'localhost'
PORT = 4223
UID = 'XYZ' # Change to your UID

ipcon = IPConnection.new # Create IP connection
l = BrickletLine.new UID, ipcon # Create device object

ipcon.connect HOST, PORT # Connect to brickd
# Don't use device before ipcon is connected

# Get threshold callbacks with a debounce time of 1 second (1000ms)
l.set_debounce_period 1000

# Register threshold reached callback for reflectivity greater than 2000
l.register_callback(BrickletLine::CALLBACK_REFLECTIVITY_REACHED) do |reflectivity|
  puts "Reflectivity: #{reflectivity}"
end

# Configure threshold for "greater than 2000"
l.set_reflectivity_callback_threshold '>', 2000, 0

puts 'Press key to exit'
$stdin.gets
ipcon.disconnect
